teamscale_client/client.py

```diff
@@ -334,6 +344,79 @@ def add_baseline(self, baseline):
         return self.put(service_url, parameters={}, data=to_json(baseline))
 
+    def get_projects(self):
+        """Retrieves a list of projects from the server.
+
+        Returns:
+            List[:class:`data.ProjectInfo`]): The list of projects.
+
+        Raises:
+            ServiceError: If anything goes wrong
+        """
+        service_url = self.get_global_service_url("projects")
+        parameters = {
+            "detail": True
+        }
+        headers = {'Accept': 'application/json'}
+        response = requests.get(service_url, params=parameters, auth=self.auth_header, verify=self.sslverify,
+                                headers=headers, timeout=self.timeout)
+        if response.status_code != 200:
+            raise ServiceError("ERROR: GET {url}: {r.status_code}:{r.text}".format(url=service_url, r=response))
+        return [
+            ProjectInfo(project_id=x['id'], name=x['name'], description=x['description'],
+                        creation_timestamp=x['creationTimestamp'], alias=x.get('alias'),
+                        deleting=x['deleting'], reanalyzing=x['reanalyzing']) for x in response.json()]
+
+    def add_project(self, project_configuration, only_config_update=False):
+        """Adds a project to Teamscale. Re-adding a project with an existing id will update the original project
+        if `only_config_update` is set to `True`.
+
+        Args:
+            project_configuration (data.ProjectConfiguration): The project that is to be added (or updated)
+            only_config_update (bool): Whether the project configuration should only be updated
+        Returns:
+            requests.Response: object generated by the upload request
+
+        Raises:
+            ServiceError: If anything goes wrong
+        """
+        service_url = self.get_global_service_url("create-project")
+        parameters = {
+            "only-config-update": only_config_update
+        }
+        response = self.put(service_url, parameters=parameters, data=to_json(project_configuration), )
+
+        if TeamscaleClient._response_indicates_failure(response):
+            raise ServiceError(
+                "ERROR: GET {url}: {status_code}:{message}".format(url=service_url, status_code=response.status_code,
+                                                                   message=TeamscaleClient._get_response_message(
+                                                                       response)))
+        return response
+
+    @staticmethod
+    def _response_indicates_failure(response):
+        """Returns a value that indicates whether the given response signals a failed request.
+
+        Args:
+            response (requests.Response): The server response.
+
+        Returns:
+            A value that indicates whether the given response signals a failed request.
+        """
+        return response.status_code != 200 or TeamscaleClient._get_response_message(response) != 'success'
```

Comment:
I opted for removing that method. You are right, this might be misleading to future users of the method. As it was used in two places only I decided to kick it.